src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -95,13 +108,39 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
       return;
     }
 
-    /*@Override
-    public void doGet() { // read
-
-
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException { // read
+        Query<Entity> query =
+           Query.newEntityQueryBuilder().setKind("users").setOrderBy(OrderBy.desc("name")).build();
+        QueryResults<Entity> results = datastore.run(query);  
+
+        List<User> users = new ArrayList<>();
+        while (results.hasNext()) {
+          Entity entity = results.next();
+
+          String u_name = entity.getString("name");
+          String u_email = entity.getString("email");
+          String u_wishlist = entity.getString("wishlist");
+          String u_teachlist = entity.getString("teachlist");
```

Comment:
Store and get list values:
```suggestion
          String u_wishlist = entity.getList("wishlist");
          String u_teachlist = entity.getList("teachlist");
```

This'll allow easier filtering on list values